content/livesync/Setup

```diff
@@ -0,0 +1,190 @@
+
+The Realtime Data Models SDK is available as an "NPM module":#npm.
+
+h3(#npm1). Use NPM
+
+Install the Ably JavaScript SDK and the Realtime Data Models SDK:
+
+```
+npm install ably @ably-labs/models
+```
+
+
+h3(#npm-link). Use the NPM link
+
+Clone this repository and run @npm link@:
+
+```
+git clone git@github.com:ably-labs/models.git
+cd models
+npm link
+```
+
+From your project, link to the cloned project and build:
+
+```
+cd ./your/project
+npm link @ably-labs/models
+pushd ./node_modules/@ably-labs/models
+npm run build
+popd
+```
+
+You should now be able to import @@ably-labs/models@ in your project.
+
+h2(#instantiate). Instantiate
+
+To instantiate the Realtime Data Models SDK, create an "Ably client":/getting-started/setup?lang=javascript and pass it into the @ModelsClient@ constructor:
+
+```
+import ModelsClient from '@ably-labs/models';
+import { Realtime } from 'ably';
+
+const ably = new Realtime.Promise({ key: 'YOUR_ABLY_API_KEY' });
+const modelsClient = new ModelsClient({ ably });
+```
+
+h2(#create-model). Create a model
+
+A @model@ represents a live, observable data model supported by the database.
+
+To create a model, you need to:
+
+1. Define the model's data structure in the frontend application.
+2. Initialize the model.
+3. Update the model based on events from the backend.
+4. Determine how end-users can modify the model.
+
+h3(#structure). Ably Realtime Data Models structure
+
+The structure of the models data as it is used in the frontend application:
+
+```
+type Post = {
+  id: number;
+  text: string;
+  comments: string[];
+}
+```
+
+h3(#initialise). Initialise
+
+An example function used by the model to initialise with the correct data from your backend:
+
+```
+async function sync() {
+  const result = await fetch('/api/post');
+  return result.json();
+};
+```
+
+h3(#merge). Merge change event
+
+The function used by the model to merge a change event that is received and the existing model state:
+
+```
+async function merge(state: Post, event: OptimisticEvent | ConfirmedEvent) {
+  return {
+    ...state,
+    text: event.data, // replace the previous post text field with the new value
+  }
+};
+```
+
+h3(#mutate). Mutate model data
+
+A function that you can use to mutate the model data in your backend:
+
+```
+async function updatePost(mutationID: string, content: string) {
+  const result = await fetch(`/api/post`, {
+    method: 'PUT',
+    body: JSON.stringify({ mutationID, content }),
+  });
+  return result.json();
+};
+```
+
+h3(#new-model-instance). Create a new model instance
+
+Create a new model instance called @post@ by passing the sync and merge functions:
+
+```
+const model = modelsClient.models.get({
+  name: 'post',
+  channelName: 'models:posts',
+  sync: sync,
+  merge: merge,
+});
+```
+
+h3(#live-changes). Subscribe to live changes
+
+Subscribe to live changes to the model data:
+
+```
+model.subscribe((err, post) => {
```

Comment:
We can also subscribe to confirmed changes only using the subscription params: https://github.com/ably-labs/models/blob/main/docs/usage.md#subscriptions

Reply:
I know the docs linked about throw the error but might be better to avoid that here as in reality you would want to handle the error. So we can just say e.g.:

```
model.subscribe((err, post) => { /* ... */ });
```

Reply:
@mschristensen will address as part of the above